src/AutoRest.CSharp/Common/Output/Models/Types/OutputLibrary.cs

```diff
@@ -10,5 +11,8 @@ internal abstract class OutputLibrary
     {
         public abstract CSharpType FindTypeForSchema(Schema schema);
         public abstract CSharpType? FindTypeByName(string originalName);
+
+        public abstract CSharpType ResolveEnum(InputEnumType enumType);
+        public abstract CSharpType ResolveModel(InputModelType model);
```

Comment:
Can we add some document to explain what these method are for?
And would these two methods be a temporary solution during the phase that we have both modeler model and cadl model? Or this is permenant?